### ISSUE TITLE
Track content type directly in the input spec

### ISSUE THREAD

**Issue by delvedor** [elastic/elastic-client-generator#297]
Currently, the input specification does not contain any information about the body content type, we assume that is a plain JSON. In some cases, it should be ndjson, which we normally represent as an array of objects.

For example, here's the bulk API:

https://github.com/elastic/elastic-client-generator/blob/ba32e0ee31ba6d2054903f10c6dd586689d7e7e9/specification/specs/document/multiple/bulk/BulkRequest.ts#L43

The only way for language generators for knowing this information is to read the `contentType` field of the `Endpoint` definition, which we are gathering from the rest-api-spec. For example:

```jsoni
"contentType": [
   "application/x-ndjson"
]
```

## 🚀 Feature Proposal

Create a new js doc tag to track this information directly in the input spec.

## Motivation

The rest-api-spec will not be there forever, and overtime we'll move most of the rest-api-spec information directly in the input spec. Furthermore I think strongly type languages will benefit from having this information easily accessible, but let's discuss more about it.

## Example

```diff
 /**
  * @rest_spec_name bulk
  * @since 0.0.0
  * @stability stable
+ * @content_type ndjson
  */
 interface BulkRequest<TSource> extends RequestBase {
   path_parts?: {
     index?: IndexName
     type?: Type
   }
   query_parameters?: {
     pipeline?: string
     refresh?: Refresh
     routing?: Routing
     _source?: boolean
     _source_excludes?: Fields
     _source_includes?: Fields
     timeout?: Time
     type_query_string?: string
     wait_for_active_shards?: WaitForActiveShards
     require_alias?: boolean
   }
   body?: Array<BulkOperationContainer | TSource>
 }
```

**Comment by swallez** [comment]
@delvedor and I discussed this, and the motivation is to make sure that ndjson is modelled as an array.

This is something that should be done in the model validation, where we have the rest-api-specs available.

There is a broader discussion however about capturing all of rest-api-specs in the input specification, and not only the content-type. In the discussion we considered introducing `Endpoint` classes alongside request and responses to capture all parts of the rest-api-specs that do not belong to the request data structure (i.e. paths, content-types, accept, methods, etc)

**Comment by delvedor** [comment]
>  In the discussion we considered introducing Endpoint classes alongside request and responses to capture all parts of the rest-api-specs that do not belong to the request data structure (i.e. paths, content-types, accept, methods, etc)

Related discussion: https://github.com/elastic/elastic-client-generator/issues/302

**Comment by JoshMock** [comment]
Presumably the ESLint rule here would be to warn when any class that extends `RequestBase` or `ResponseBase` does not include a `@content_type`.

Although, in almost every case the value is going to be `application/json`. So maybe it shouldn't be the subject of a linting rule, and all spec compilers should automatically add a JSON content type annotation by default if none is provided?

**Comment by pquentin** [comment]
Agreed, ESLint won't be of help here. But we do want to capture non-JSON Content-Types, at least for docs.
